pkg/replication/order.go

```diff
@@ -0,0 +1,110 @@
+package replication
+
+import (
+	"sort"
+
+	"github.com/square/p2/pkg/health"
+	"github.com/square/p2/pkg/util"
+)
+
+type Order interface {
+	// Execute initiates the rollout order. It runs in a blocking loop. The
+	// next node to update should be received from nextCh. After a node is done
+	// updating, it should be returned on doneCh. When the rollout is done,
+	// Execute should close nextCh and then terminate.
+	// errCh is used to report errors. Sending on quitCh will terminate Execute.
+	Execute(nextCh chan<- string, doneCh <-chan string, errCh chan<- error, quitCh <-chan struct{})
+
+	// Nodes returns the list of all nodes that the Order will cover. The list
+	// is in arbitrary order.
+	Nodes() []string
+}
+
+type MaxOrder struct {
+	active int
+	nodes  []string
+}
+
+// create a new MaxOrder using the given node list. The maximum number of
+// concurrent updates permitted by this order is active. For example, if
```

Comment:
```
The maximum number ofconcurrent updates permitted by this order is active.
```
This was really confusing.  You mean "the parameter `active` designates the maximum number of concurrent updates"